File: c/isogram/src/isogram.c
#include "isogram.h"
#include <stdio.h>
#include <ctype.h>
#include <string.h>

bool is_isogram(const char* str){
    if(str == NULL){
        return 0;
    }
    int string_length = strlen(str);
    if(string_length == 0){
        return 1;
    }
    for (int i = 0; i < string_length; ++i){
        char c = tolower(str[i]);
        for (int j = 0; j < string_length; ++j){
            if(j!=i){
                char e = tolower(str[j]);
                if( (isalnum(c) && isalnum(e)) && c == e){
                    return 0;
                }
            }
        }
    }
    return 1;
}

File: erlang/isogram/src/isogram.erl
%% @doc
%%   Determine if a word or phrase is an isogram.
%%   An isogram (also known as a "nonpattern word") is a word or phrase without a repeating letter, however spaces and hyphens are allowed to appear multiple times.
%%   Examples of isograms:
%%   lumberjacks
%%   background
%%   downstream
%%   six-year-old
%%   The word isograms, however, is not an isogram, because the s repeats.
%% @end

-module(isogram).

-export([
    is_isogram/1,
    test_version/0
]).

%% @doc
%%    is_isogram takes in a list of characters ( aka string ) and returns a boolean
%%    value that indicates whether a string passes the isogram criteria.
%% @end
is_isogram(String) ->
    is_isogram(String, []).

is_isogram([], _) ->
    true;
is_isogram([ H | T ], R) when
        H =:= $  orelse
        H =:= $- ->
    is_isogram(T, R);
is_isogram([ H | T ], R) ->
    LH = string:to_lower(H),
    case lists:member(LH, R) of
        true ->
            false;
        false ->
            is_isogram(T, [ LH | R ])
    end.

test_version() ->
    1.
